### PR TITLE
Expand attribute functionality

### PR DESCRIPTION
**Pull request by letterix**
closes #45
closes #46

### CODE REVIEW

File: app/routes/attribute.routes.js [hunk 1]
```diff
@@ -22,5 +22,22 @@ module.exports = function(routes) {
...
+        attributeController.getAttributeById(request.params.id)
+            .then(responseHandler.sendJsonResponse(response))
+            .catch(responseHandler.sendErrorResponse(response));
+    });
+
+    // delete an attribute given an id
+    routes.delete('/:id', function(request, response) {
+        attributeController.deleteAttributeById(request.params.id)
+            .then(function() {
+                return response.json({ message: 'The attribute were successfully removed.' });
```
Comment (Tobbe) [inline app/routes/attribute.routes.js]: couldn't you put this in responseHandler as well?

File: app/routes/role.routes.js [hunk 2]
```diff
@@ -33,7 +33,9 @@ module.exports = function(routes) {
     // delete a role given an id
     routes.delete('/:id', function(request, response) {
         roleController.deleteRoleById(request.params.id)
-            .then(responseHandler.sendJsonResponse(response))
+            .then(function() {
+                return response.json({ message: 'The role were successfully removed.' });
```
Comment (Tobbe) [inline app/routes/role.routes.js]: put in responseHandler?

**Comment by letterix** [comment]
well, we can. And too be honest we should :)

**Comment by letterix** [comment]
Long, but self-explanatory; which is more profitable then the reverse. Changing the name.

File: app/routes/role.routes.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ module.exports = function(routes) {
     // delete a role given an id
     routes.delete('/:id', function(request, response) {
         roleController.deleteRoleById(request.params.id)
-            .then(responseHandler.sendJsonResponse(response))
+            .then(responseHandler.sendSuccessfulDeleteJsonResponse(response))
```
Comment (Tobbe) [inline app/routes/role.routes.js:36]: This has to be wrong. You're invoking the function directly here, which should result in directly sending the response

File: app/utils/response.handler.js [hunk 4]
```diff
@@ -91,4 +96,8 @@ exports.sendJsonResponse = function(response) {
     return function(object) {
         return response.json(object);
     }
-};
\ No newline at end of file
+};
+
+exports.sendSuccessfulDeleteJsonResponse = function(response) {
+    response.json({ message: 'The item was successfully removed.' });
```
Comment (Tobbe) [inline app/utils/response.handler.js:102]: This should return a function, just like `sendJsonResponse` does

**Comment by letterix** [comment]
hmm, testing right now. But it seems to be working fine. I see your point though and would also agree on that it shouldn't work like this. however the result is status 200 with {"message": "The item was successfully removed."} as a body. Any thoughts?

**Comment by letterix** [comment]
The method needed to return a function or it would execute without waiting for the value from the promise.

**Pull request merged**